### PR TITLE
RIA-4722 Process to enable correct state to be set when clarifying questions is completed

### PR DESCRIPTION
**Pull request by srinivasuluputcha**
:

###https://tools.hmcts.net/jira/browse/RIA-4722 ###

### Clarifying question: Process to enable correct state to be set when clarifying questions is completed###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/CompleteClarifyingQuestionsHandler.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        Optional<List<IdValue<Direction>>> directionsListOptional = asylumCase.read(AsylumCaseFieldDefinition.DIRECTIONS);
+
+        Optional<IdValue<Direction>> clarifyingQuestionsDirectionOptional = directionsListOptional.orElse(Collections.emptyList()).stream()
+            .filter(directionIdVale -> directionIdVale.getValue().getTag().equals(DirectionTag.REQUEST_CLARIFYING_QUESTIONS))
+            .findFirst();
```
Comment (jkumar-moj) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/CompleteClarifyingQuestionsHandler.java:57]: Is it ok to pick the first one? Will it always be sorted and has it been tested with multiple rounds of clarifying questions?

**Review by jkumar-moj — CHANGES_REQUESTED** [review]
@srinivasuluputcha good one. looks good couple of minor comments to ensure it is already covered and minor change in the functional test. Could you please take a look and confirm?

File: src/functionalTest/resources/scenarios/RIA-4722-direction-request-clarifying-questions.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                  "value": {
+                    "question": "Question 1"
+                  }
+                }
+              ]
+            }
+          }
+        ],
+        "sendDirectionActionAvailable": "No",
+        "preClarifyingState": "reasonsForAppealSubmitted"
```
[the remark this answers is not in the document]
Comment (srinivasuluputcha) [reply src/functionalTest/resources/scenarios/RIA-4722-direction-request-clarifying-questions.json]: I've fixed this one.

**Review by jkumar-moj — APPROVED** [review]
👍

**Review by bradsorour — APPROVED** [review]
LGTM

**Pull request merged**